infrastructure/analytics/pact_test.go

```diff
@@ -0,0 +1,103 @@
+package analytics
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"log"
+	"net/http"
+	"testing"
+
+	"github.com/pact-foundation/pact-go/dsl"
+
+	"github.com/snyk/snyk-ls/internal/testutil"
+)
+
+func TestAnalyticsProviderPact(t *testing.T) {
+	testutil.NotOnWindows(t, "we don't have a pact cli")
+
+	pact := &dsl.Pact{
+		Consumer: "snyk-ls",
+		Provider: "AnalyticsProvider",
+		LogDir:   "logs",
+		PactDir:  "./pacts",
+		LogLevel: "DEBUG",
+	}
+
+	defer pact.Teardown()
+
+	var test = func() (err error) {
+		u := fmt.Sprintf("http://localhost:%d/rest/api/orgs/org_id/analytics", pact.Server.Port)
+
+		expectedBody := getExpectedBodyRequest()
+		bodyBytes, err := json.Marshal(expectedBody)
+
+		if err != nil {
+			fmt.Printf("Error marshaling payload: %v\n", err)
+			return err
+		}
+
+		req, err := http.NewRequest("POST", u, bytes.NewBuffer(bodyBytes))
+		if err != nil {
+			return err
+		}
+
+		req.Header.Set("Content-Type", "application/json")
+		if _, err = http.DefaultClient.Do(req); err != nil {
+			return err
+		}
+
+		return nil
+	}
+
+	pact.
+		AddInteraction().
+		Given("Analytics data is ready").
+		UponReceiving("A request to create analytics data").
+		WithRequest(dsl.Request{
+			Method:  "POST",
+			Path:    dsl.String("/rest/api/orgs/org_id/analytics"),
+			Headers: dsl.MapMatcher{"Content-Type": dsl.Term("application/json", `^application\/json$`)},
+			Body:    getExpectedBodyRequest(),
+		}).
+		WillRespondWith(dsl.Response{
+			Status:  201,
+			Headers: dsl.MapMatcher{"Content-Type": dsl.Term("application/json", `^application\/json$`)},
+			Body:    map[string]interface{}{},
+		})
```

Comment:
@bastiandoetsch , `AddInteraction` is needed so the mock service have a reference to validate the incoming requests. This enables `pact.Verify(test)` to confirm if our actual HTTP request meets the expectations set by `AddInteraction`.